composables/api/useUserProfileUpdate.ts

```diff
@@ -0,0 +1,45 @@
+import { useMutation } from 'vue-query';
+import type { Database } from '~/types/generated-types';
+
+export const useUserProfileUpdate = () => {
+  const client = useSupabaseClient<Database>();
+  const user = useSupabaseUser();
+
+  const result = useMutation(
+    async (profileData: { profileName?: string; file?: File | null }) => {
+      if (!user.value?.id) {
+        console.warn('You are not allowed to perform this action.');
+        return;
+      }
+
+      const { profileName, file } = profileData;
+      let licensePath;
+
+      if (file) {
+        const filePath = `${user.value.id}/${file.name}`;
+        const { data: uploadData, error: uploadError } = await client.storage
+          .from('license-files')
+          .upload(filePath, file, { upsert: true });
+
+        if (uploadError) {
+          console.error(uploadError);
+          throw new Error(uploadError.message);
+        }
+
+        licensePath = uploadData.path;
+      }
```

Comment:
how it works? 👀

Reply:
I've created a bucket in our Supabase Storage called `licence-files`. To the bucket I'm uploading the file provided by our frontend application. To do this, I'm creating a path of a personal folder (name of the folder is the same as the user ID" to put there our file. `{ upsert: true }` option means that if the file with provided name exist, we just overwrite it (it's an easier logic than creating new names for the file, I would have to use much more methods for that including listing the files from the bucket).

<img width="1490" alt="image" src="https://user-images.githubusercontent.com/66744401/228523477-0dfa348d-4025-4472-a5d5-4b1285b8e4f8.png">


Afterwards, when the API responds, I saved the `licencePath` and then passed it to certain user profile in our database:
<img width="913" alt="image" src="https://user-images.githubusercontent.com/66744401/228523300-7c243a72-6fdb-4370-a2e7-538a52ac79cb.png">

I also had to configure some RLS policies for it to not throw errors. More shortly explained here: https://supabase.com/docs/reference/javascript/storage-from-upload

(btw I love these docs, they teach you so fast 💚 )